sockeye/inference.py

```diff
@@ -706,6 +707,32 @@ def __init__(self,
             self.strip_ids.add(self.unk_id)
         self.models = models
 
+        # interleave qkv with heads for fast attention inference
+        for model in self.models:
+            for layer in [l for l in model.encoder.layers] + [l for l in model.decoder.layers]:
```

Comment:
could you add some comments adding shape information for easier understanding?
I wonder, if this can be moved to the `SockeyeModel` class instead. It has an `inference_only` flag to control behavior for some speed optimizations. WOuld it be possible to trigger the interleaving below based on this flag, and move this code there? It could be a separate method similar to `disable_dropout`, see https://github.com/awslabs/sockeye/blob/sockeye_2/sockeye/model.py#L442

Reply:
This part of the code is intended to be only temporary until I have training working with these optimizations. The reason is that the `interleaved_matmul_encdec` ops for the faster multi-head attention assume a different structure to the key_value projections, which I address by modifying the projection weights after initializing the model. This shouldn't be needed anymore once models can be trained with these ops.

Reply:
Actually, one reason to retain this code is for compatibility with older models, which would then also require a model version check. If that's the case, this can be moved to a more appropriate location.

Reply:
If we want to merge this PR rather soon, we should maintain backwards compatibility and in that case a more appropriate location would be preferred by me. 
Once this optimization is available for training as well, we can do a major version bump (2.1).